tests/unit/test_extensionPoints.py

```diff
@@ -0,0 +1,255 @@
+#tests/unit/test_extensionPoints.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""Unit tests for the extensionPoints module.
+"""
+
+import unittest
+import extensionPoints
+
+class ExampleClass(object):
+	def method(self):
+		return 42
+
+def exampleFunc():
+	return 3.14
+
+class TestBoundMethodWeakref(unittest.TestCase):
+
+	def onDelete(self, weak):
+		self.deleted = weak
+
+	def setUp(self):
+		self.deleted = None
+		self.instance = ExampleClass()
+		self.weak = extensionPoints.BoundMethodWeakref(self.instance.method, self.onDelete)
+
+	def test_get(self):
+		"""Test that we get the right strong reference from the weak reference.
+		"""
+		method = self.weak()
+		self.assertEqual(method, self.instance.method)
+
+	def test_onDelete(self):
+		"""Test that the deletion callback gets called.
+		"""
+		del self.instance
+		self.assertEqual(self.deleted, self.weak)
+
+	def test_isWeak(self):
+		"""Test that this is actually a weak reference;
+		i.e. that it dies when the instance dies.
+		"""
+		del self.instance
+		method = self.weak()
+		self.assertIsNone(method)
+
+class TestCallWithSupportedKwargs(unittest.TestCase):
+
+	def test_supportsNoKwargs(self):
+		called = []
+		def h():
+			called.append(True)

```

Comment:
> I'm surprised this isn't an int being incremented?

No can do. The closure captures "called", and because it's a list, we can mutate the value itself. In contrast, an int is immutable. The only way to increment an int is to assign a new value to "called", but if we do this, this only affects the closure; the outer scope (the test itself where the assertion happens) won't get the new value of "called".